bouter/utilities.py

```diff
@@ -93,6 +93,168 @@ def extract_segments_above_threshold(
 
     return np.array(segments), np.array(connected)
 
+@jit(nopython=True)
+def get_score_trace(trace_pad, kernel, bias=-0.2839):
+    """
+    Used to calculate the correlation score between the squared velocity
+    trace and the kernel for bout detection.
+    :param trace_pad: squared velocity (optionally) padded
+    :param kernel: kernel used for bout detection
+    :return:
+        array of correlation scores that matches with the trace by index.
+    """
+    
+    # For numerical stability.
+    eta = 0.0000000001
+    
+    corr = np.empty((len(trace_pad) - len(kernel)))
+    for i in range(len(trace_pad) - len(kernel)):
+        current_values = trace_pad[i:i+len(kernel)]
+        relative_values = current_values / (np.max(current_values) + eta)
+        
+        # Simplified convolution, works faster and with Numba.
+        conv = np.sum(relative_values * np.flip(kernel))
+        
+        corr[i] = 1 / (1 + np.exp(conv + bias))
+        
+    # Align the correlation score trace with the actual bouts.
+    # Found by trial and error - so can be changed as needed.
+    corr = corr[:-int(kernel.shape[0]/5)]
+    corr = np.concatenate((np.zeros(int(kernel.shape[0]/5)), corr))
+        
+    return (corr - 1) * -1
+
+
+def calc_bout_score(trace, kernel=None, bias=-0.2839, pad_len=None):
+    """ 
+    Wrapper for the get_score_trace function.
+    :param trace: the trace to detect bouts in.
+    :param kernel: kernel used for bout detection.
+    :param bias: the bias to shift the output
+    :param pad_len: the length of the padding on each side.
+    :return:
+        array of correlation scores that matches with the trace by index.
+    """
+
+    if kernel is None:
```

Comment:
An issue I see with this approach is that the kernel is probably quite dependent on the rate at what your data has been sampled. One could up/downsample this kernel, but then it should at least be indicated the sampling rate of the data it is based on, and instead of hardcoded in the function, maybe it could be stored in an assets/kernels/whatever folder from where one can load it and modify it before using it, rather than just having to alter it within the function. 
Maybe (big maybe) adding a function/instruction on how to generate one's own?